stor/dx.py

```diff
@@ -496,7 +731,8 @@ def virtual_path(self):
         proj = dxpy.DXProject(dxid=self.project)
         virtual_p = DXVirtualPath(self.drive + proj.name + ':/')
         if self.resource:
-            file_h = dxpy.DXFile(dxid=self.canonical_resource)
+            file_h = dxpy.DXFile(dxid=self.canonical_resource,
+                                 project=self.canonical_project)
             virtual_p = virtual_p / file_h.folder[1:] / file_h.name
```

Comment:
with all these cached properties, if you call move / remove / rename, do you then need to then clear the cache?

Reply:
I could do this, or I could do the wrapper mentioned in your summary which would take care of the scenarios you had listed. I prefer doing the latter. Should I be doing both?

Reply:
def just do the latter, but still need to keep in mind clearing caches after moving (can open JIRA ticket to track this)